app/Platform/Commands/SyncEvents.php

```diff
@@ -0,0 +1,2199 @@
+                [485510, 485516, 485522, 485528],
+                noWinners: true,
+            ),
+        ];
+
+        $id = $this->argument('gameId');
+        if ($id) {
+            if (!array_key_exists($id, $gameConversions)) {
+                $this->error("No conversion defined for game $id");
+
+                return;
+            }
+
+            $gameConversions = [
+                $id => $gameConversions[$id],
+            ];
+
+            // when processing a single game, run all the jobs synchronously so the validation
+            // can check the final state.
+            config(['queue.default' => 'sync']);
+        }
+
+        $gameCount = count($gameConversions);
+
+        $test = $this->option('test');
+        if ($test) {
+            $this->info("\nTesting conversion of {$gameCount} event games.");
+        } else {
+            $this->info("\nUpserting {$gameCount} events derived from event games.");
+        }
+        $progressBar = $this->output->createProgressBar($gameCount);
+
+        $before = [];
+        foreach ($gameConversions as $gameId => $conversion) {
+            if ($test) {
+                $before = $conversion->captureBefore($gameId);
+                DB::beginTransaction();
+            }
+
+            try {
+                $conversion->convert($this, $gameId);
+
+                if ($test) {
+                    $this->dumpConverted($this, $gameId);
+                    $conversion->validate($this, $gameId, $before);
+                }
+            } catch (Exception $e) {
+            }
+
+            if ($test) {
+                DB::rollBack();
+            }
+
+            $progressBar->advance();
+        }
+
+        $progressBar->finish();
+
+        $this->info("\nDone.");
+    }
+
+    private function dumpConverted(Command $command, int $gameId): void
+    {
+        $game = Game::find($gameId);
+        $event = $game->event;
+
+        $achievements = $game->achievements()->published()->orderBy('DisplayOrder')->get();
+        foreach ($achievements as $achievement) {
+            $command->info(sprintf("  %6u %5u/%u %s", $achievement->ID, $achievement->unlocks_hardcore_total, $game->players_hardcore, $achievement->Title));
+            if ($achievement->Points === 0) {
+                $command->error("No points on achievement {$achievement->ID}");
+            } elseif ($achievement->unlocks_hardcore_total > $game->players_hardcore) {
+                $command->error("Too many achievements awarded for {$achievement->ID}");
+            }
+        }
+    }
+}
+
+// Base class - provides helper functions used by the individual converters below.
+class ConvertGame
+{
+    protected string $slug;
+    protected ?string $activeFrom = null;
+    protected ?string $activeThrough = null;
+    protected bool $noWinners = false;
+
+    public function convert(Command $command, int $gameId): void
```

Comment:
Still in the process of reviewing, but wanted to fire one comment off before a review is complete.

I'm wondering if this function in particular should leverage transactions, ie:
```php
public function convert(...): void
{
  DB::beginTransaction();

  try {
    $game = Game::find($gameId);
     // ... stuff ...

    $this->process($command, $event);
    DB::commit();
  } catch (Exception $e) {
    DB:rollBack();
    // add some logging
  }
}
```

This way in the off-chance something goes wrong in the prod transformations, it's easier to recover from without manual intervention. This might make sense given the transformations are both complex and unidirectional.

Reply:
Convert is already wrapper in a transaction, but only in `--test` mode:
```
            if ($test) {
                $before = $conversion->captureBefore($gameId);
                DB::beginTransaction();
            }

            try {
                $conversion->convert($this, $gameId);

                if ($test) {
                    $this->dumpConverted($this, $gameId);
                    $conversion->validate($this, $gameId, $before);
                }
            } catch (Exception $e) {
            }

            if ($test) {
                DB::rollBack();
            }
```
I could expand that for non-test mode too.